examples/closure-app/linux/ClosureManager.cpp

```diff
@@ -0,0 +1,267 @@
+{
+    // Add logic to handle the Step command
+    return Status::Success;
+}
+
+void ClosureManager::HandleClosureActionTimer(System::Layer * layer, void * aAppState)
+{
+    // Mark aAppState as unused to avoid compiler warnings
+    // Will be used in closure dimension cluster Commands
+    (void) aAppState;
+
+    ClosureManager & instance = ClosureManager::GetInstance();
+
+    switch (instance.mCurrentAction)
+    {
+    case ClosureAction::kCalibrateAction:
+        instance.HandleClosureActionComplete(ClosureAction::kCalibrateAction);
+        break;
+    case ClosureAction::kStopAction:
+        // Add logic to handle Stop action completion
+        break;
+    case ClosureAction::kMoveToAction:
+        // Add logic to handle MoveTo action completion
+        break;
+    case ClosureAction::kLatchAction:
+        // Add logic to handle Latch action completion
+        break;
+    case ClosureAction::kSetTargetAction:
+        // Add logic to handle SetTarget action completion
+        break;
+    case ClosureAction::kStepAction:
+        // Add logic to handle Step action completion
+        break;
+    case ClosureAction::kPanelLatchAction:
+        // Add logic to handle Panel Latch action completion
+        break;
+    default:
+        ChipLogError(AppServer, "Invalid action received in HandleClosureActionTimer");
+        break;
+    }
+}
+
+void ClosureManager::HandleClosureActionComplete(ClosureAction action)
+{
+    ChipLogProgress(AppServer, "HandleClosureActionComplete called for action: %d", static_cast<int>(action));
+
+    switch (action)
+    {
+    case ClosureAction::kCalibrateAction: {
+        mClosureEndpoint1.OnCalibrateActionComplete();
+        mClosurePanelEndpoint2.OnCalibrateActionComplete();
+        mClosurePanelEndpoint3.OnCalibrateActionComplete();
+        mIsCalibrationActionInProgress = false;
+        break;
+    }
+
+    case ClosureAction::kStopAction: {
+        if (mIsCalibrationActionInProgress)
+        {
+            mClosureEndpoint1.OnStopCalibrateActionComplete();
+            mClosurePanelEndpoint2.OnStopCalibrateActionComplete();
+            mClosurePanelEndpoint3.OnStopCalibrateActionComplete();
+            mIsCalibrationActionInProgress = false;
+        }
+        else if (mIsMoveToActionInProgress)
+        {
+            mClosureEndpoint1.OnStopMotionActionComplete();
+            mClosurePanelEndpoint2.OnStopMotionActionComplete();
+            mClosurePanelEndpoint3.OnStopMotionActionComplete();
+            mIsMoveToActionInProgress = false;
+        }
+        else if (mIsSetTargetActionInProgress)
+        {
+            // Add logic to handle stopping SetTarget action
+            mIsSetTargetActionInProgress = false;
+        }
+        else if (mIsStepActionInProgress)
+        {
+            // Add logic to handle stopping Step action
+            mIsStepActionInProgress = false;
```

Comment:
I'm not convinced we need a stop function for each type of command - the stop behavior should be the irrespective of the command being stoped but this can be addressed in a follow up.